src/ddb/tests/ddb_commands_print_tests.c

```diff
@@ -0,0 +1,331 @@
+/**
+ * (C) Copyright 2022 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+#include <ddb_vos.h>
+#include <ddb_printer.h>
+#include "ddb_cmocka.h"
+#include "ddb_test_driver.h"
+
+static struct ddb_ctx ctx = {.dc_io_ft.ddb_print_message = dvt_fake_print};
+
+#define assert_str_exact(str) assert_string_equal(str, dvt_fake_print_buffer)
+#define assert_str(str) assert_string_contains(dvt_fake_print_buffer, str)
+
+static void
+print_container_test(void **state)
+{
+	struct ddb_cont cont = {0};
+
+	uuid_parse("12345678-1234-1243-1243-123456789012", cont.ddbc_cont_uuid);
+	cont.ddbc_idx = 1;
+
+	ddb_print_cont(&ctx, &cont);
+	assert_str_exact("[1] 12345678-1234-1243-1243-123456789012\n");
+}
+
+static void
+print_object_test(void **state)
+{
+	struct ddb_obj obj = {0};
+
+	obj.ddbo_idx = 2;
+	obj.ddbo_oid.lo = 1;
+	obj.ddbo_oid.hi = 10;
+	obj.ddbo_nr_grps = 2;
+	strcpy(obj.ddbo_obj_class_name, "TEST CLASS");
+	strcpy(obj.ddbo_otype_str, "TEST TYPE");
+
+	ddb_print_obj(&ctx, &obj, 1);
+
+	assert_str_exact("\t[2] '10.1' (class: TEST CLASS, type: TEST TYPE, groups: 2)\n");
+}
+
+static void set_key_buf(struct ddb_key	*key, uint32_t len)
+{
+	int i;
+
+	for (i = 0; i < len; i++)
+		((uint8_t *)key->ddbk_key.iov_buf)[i] = (i % 16 + 0x1);
+
+	key->ddbk_key.iov_len = len;
+}
+
+static void
+print_key_test(void **state)
+{
+	struct ddb_key	key = {0};
+	char		key_buf[1024] = {0};
+
+	key.ddbk_idx = 4;
+	d_iov_set(&key.ddbk_key, key_buf, ARRAY_SIZE(key_buf));
+
+	ddb_print_key(&ctx, &key, 2);
+
+	/* empty large key */
+	assert_str_exact("\t\t[4] '' (1024)\n");
+	dvt_fake_print_reset();
+
+	/* Large key buffer, but only part is text */
+	strcpy(key_buf, "string key");
+	ddb_print_key(&ctx, &key, 0);
+	assert_str_exact("[4] 'string key' (1024)\n");
+	dvt_fake_print_reset();
+
+	/* No ending '\0' */
+	strcpy(key_buf, "abcdefghijklmnopqrstuvwxyz");
+	key.ddbk_key.iov_len = 5;
+	ddb_print_key(&ctx, &key, 0);
+	assert_str_exact("[4] 'abcde' (5)\n");
+	dvt_fake_print_reset();
+
+	/* With ending '\0' in middle ... only prints to null terminator */
+	strcpy(key_buf, "abcdefghijklmnopqrstuvwxyz");
+	key_buf[10] = '\0';
+	key.ddbk_key.iov_len = 26;
+	ddb_print_key(&ctx, &key, 0);
+	assert_str_exact("[4] 'abcdefghij' (26)\n");
+	dvt_fake_print_reset();
+
+	/*
+	 * Print binary keys.
+	 * If key length is a number type, then print as that.
+	 */
+	memset(key_buf, 0, ARRAY_SIZE(key_buf));
+
+	/* char key */
+	key_buf[0] = 0xab;
+	key.ddbk_key.iov_len = sizeof(char);
+	ddb_print_key(&ctx, &key, 0);
+	assert_str_exact("[4] '{uint8:0xab}'\n");
+	dvt_fake_print_reset();
+
+	/* short key */
+	short s = 0xabcd;
+	key.ddbk_key.iov_buf = (uint8_t *)&s;
```

Comment:
(style)  Missing a blank line after declarations

Reply:
any reason not to just declare these at the top to avoid these warnings?   Alternatively, you could make a macro for setting up the variable and calling the print function that takes the type as an argument